### ISSUE TITLE
Actions steps seem to use obfuscated env var value for project ID instead of actual value when project ID is an environment variable

### ISSUE THREAD

**Issue by heidi-jack** [google-github-actions/auth#290]
### TL;DR

When using an environment variable for the `project_id` input variable, instead of a string directly, this step seems to export
new environment variables with an obfuscated value.

I assume this is due to the way GitHub Actions will obfuscate the value of an environment variable with it's name when printing in logs, so you know the source of the value.

This works:
```
      - name: Gcloud Auth
        uses: google-github-actions/auth@v1
        with:
          credentials_json: ${{ secret.MY_GCP_SECRET }}
          project_id: mgcp-123456-some-project
```

This doesn't:
```
      - name: Gcloud Auth
        uses: google-github-actions/auth@v1
        with:
          credentials_json: ${{ secret.MY_GCP_SECRET }}
          project_id: ${MY_ENV_VAR}
```

Note that the auth step completes successfully, but google-github-actions/setup-gcloud@v1 uses the outputted env vars for the project ID by the looks of it, and these are set incorrectly (see logs below)

### Expected behavior

Value of env var is properly exported from GCP Auth step. 

### Observed behavior

Value of env var is not properly exported from GCP Auth step, seems to be the obfuscated version instead.

### Action YAML

```yaml
## Failing

name: Publish Artifact
on: push
  
env:
  GCP_PROJECT_ID: mgcp-123456-some-project

jobs:
  gcloud:
    runs-on: ubuntu-latest
    steps:
      - name: Gcloud Auth
        uses: google-github-actions/auth@v1
        with:
          credentials_json: ${{ secret.MY_GCP_SECRET }}
          project_id: ${GCP_PROJECT_ID}
    
      - name: GCloud Set Up SDK
        uses: google-github-actions/setup-gcloud@v1
        with:
          project_id: ${GCP_PROJECT_ID}

---

## Successful

name: Publish Artifact
on: push
  
# env:
#   GCP_PROJECT_ID: mgcp-123456-some-project

jobs:
  gcloud:
    runs-on: ubuntu-latest
    steps:
      - name: Gcloud Auth
        uses: google-github-actions/auth@v1
        with:
          credentials_json: ${{ secret.MY_GCP_SECRET }}
          project_id: mgcp-123456-some-project
    
      - name: GCloud Set Up SDK
        uses: google-github-actions/setup-gcloud@v1
        with:
          project_id: mgcp-123456-some-project
```

### Log output

```text
Failing One:


Run google-github-actions/auth@v1
  with:
    credentials_json: ***
  
    project_id: $***GCP_PROJECT_ID***
    create_credentials_file: true
    export_environment_variables: true
    cleanup_credentials: true
    access_token_lifetime: 3600s
    access_token_scopes: https://www.googleapis.com/auth/cloud-platform
    retries: 0
    id_token_include_email: false
  env:
    GCP_PROJECT_ID: mgcp-123456-some-project
Created credentials file at "creds-file-path/creds.json"

Run google-github-actions/setup-gcloud@v1
  with:
    project_id: $***GCP_PROJECT_ID***
    skip_install: false
    version: latest
  env:
    GCP_PROJECT_ID: mgcp-123456-some-project
    CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: creds-file-path/creds.json
    GOOGLE_APPLICATION_CREDENTIALS: creds-file-path/creds.json
    GOOGLE_GHA_CREDS_PATH: creds-file-path/creds.json
    CLOUDSDK_CORE_PROJECT: $***GCP_PROJECT_ID***
    CLOUDSDK_PROJECT: $***GCP_PROJECT_ID***
    GCLOUD_PROJECT: $***GCP_PROJECT_ID***
    GCP_PROJECT: $***GCP_PROJECT_ID***
    GOOGLE_CLOUD_PROJECT: $***GCP_PROJECT_ID***
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/6cbe6f44-c29d-4bea-96c1-dd8ab842b00d -f /home/runner/work/_temp/62e175ac-429a-47a8-a030-8878c74c244f
Error: google-github-actions/setup-gcloud failed with: failed to execute command `gcloud --quiet auth login --force --cred-file creds-file-path/creds.json: ERROR: (gcloud.auth.login) The project property must be set to a valid project ID, [$***GCP_PROJECT_ID***] is not a valid project ID.
```

For when we use the value direclty with the action instead of through env vars:

```
Run google-github-actions/auth@v1
  with:
    credentials_json: ***
  
    project_id: mgcp-123456-some-project
    create_credentials_file: true
    export_environment_variables: true
    cleanup_credentials: true
    access_token_lifetime: 3600s
    access_token_scopes: https://www.googleapis.com/auth/cloud-platform
    retries: 0
    id_token_include_email: false
  env:
    GCP_PROJECT_ID: mgcp-123456-some-project
Created credentials file at "creds-file-path/creds.json"

Run google-github-actions/setup-gcloud@v1
  with:
    project_id: mgcp-123456-some-project
    skip_install: false
    version: latest
  env:
    GCP_PROJECT_ID: mgcp-123456-some-project
    CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: creds-file-path/creds.json
    GOOGLE_APPLICATION_CREDENTIALS: creds-file-path/creds.json
    GOOGLE_GHA_CREDS_PATH: creds-file-path/creds.json
    CLOUDSDK_CORE_PROJECT: mgcp-123456-some-project
    CLOUDSDK_PROJECT: mgcp-123456-some-project
    GCLOUD_PROJECT: mgcp-123456-some-project
    GCP_PROJECT: mgcp-123456-some-project
    GOOGLE_CLOUD_PROJECT: mgcp-123456-some-project
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/d7db9d40-cfac-4c6e-9b2f-e2f4a3590cdb -f /home/runner/work/_temp/c999c4f1-9ce6-4d41-9cad-af092bdec40d
Successfully authenticated
Successfully set default project
```
```


### Additional information

_No response_

**Comment by sethvargo** [comment]
Hi @heidi-jack - this is how GitHub Actions YAML interpolation works - we are getting the value you directly provided. The `${FOO}` syntax only works insides shell scripts, since that is the syntax for accessing environment variables in bash, zsh, etc.; it's equivalent to `$FOO`.

GitHub Actions YAML files are not parsed as bash, so that expansion never happens. 

```yaml
- run: |-
    echo ${FOO} # works, since this is running as bash
  env:
    BAR: ${FOO} # does not work, since this is just YAML
```

To access GitHub Actions environment variables outside of a shell script, you need to use the double braces with the `env` prefix:

```yaml
- run: |-
    echo ${FOO} # works, since this is running as bash
  env:
    BAR: ${{ env.FOO }} # works, since GitHub interpolates this
```

So in your case, the project ID is being set to the literal string `"${GCP_PROJECT_ID}"`. However, because you are using a service account key JSON, GitHub Actions is obfuscating the `{` and `}` parts of the value because it thinks they are secrets (from your JSON file).

**Comment by heidi-jack** [comment]
Thanks, that was silly of me!
